lib/fetchers/PortionFetcher.ts

```diff
@@ -0,0 +1,105 @@
+import { Unit } from 'aws-embedded-metrics';
+import NodeCache from 'node-cache';
+import axios from '../providers/quoters/helpers';
+import { log } from '../util/log';
+import { metrics } from '../util/metrics';
+
+export enum PortionType {
+  Flat = 'flat',
+  Regressive = 'regressive',
+}
+
+export interface Portion {
+  readonly bips: number;
+  readonly receiver: string;
+  readonly type: PortionType;
+}
+
+export interface GetPortionResponse {
+  readonly hasPortion: true;
+  readonly portion: Portion;
+}
+
+export interface GetNoPortionResponse {
+  readonly hasPortion: false;
+}
+
+export type AllPortionResponse = GetPortionResponse | GetNoPortionResponse
+
+export const GET_NO_PORTION_RESPONSE: GetNoPortionResponse = { hasPortion: false }
+
+export class PortionFetcher {
```

Comment:
nit: would rather call this something like `PortionProvider`

Reply:
not disagreeing honestly, but curious as to when we pick between provider vs fetcher?